src/luxonis_ml/ops/dataset.py

```diff
@@ -0,0 +1,740 @@
+        """
+
+        with open(annotation_path) as file:
+            coco = json.load(file)
+
+        images = coco['images']
+        annotations = coco['annotations']
+        coco_categories = coco['categories']
+        categories = {cat['id']:cat['name'] for cat in coco_categories}
+        keypoint_definitions = {cat['id']:{'keypoints':cat['keypoints'],'skeleton':cat['skeleton']} \
+                                for cat in coco_categories if 'keypoints' in cat.keys() and 'skeleton' in cat.keys()}
+
+        if override_main_component is not None:
+            component_name = override_main_component
+        else:
+            component_name = self.sources[source_name].main_component
+
+        for image in tqdm(images):
+            new_ann = {
+                component_name: {
+                    'annotations': []
+                }
+            }
+            fn = image['file_name']
+            image_id = image['id']
+            image_path = f"{image_dir}/{fn}"
+            if os.path.exists(image_path):
+                img = cv2.imread(image_path)
+                anns = [ann for ann in annotations if ann['image_id'] == image_id]
+                for ann in anns:
+                    cat_id = ann['category_id']
+                    class_name = categories[cat_id]
+                    class_id = self._add_class(class_name)
+                    if cat_id in keypoint_definitions.keys():
+                        self._add_keypoint_definition(class_id, keypoint_definitions[cat_id])
+
+                    new_ann_instance = {}
+                    new_ann_instance['class'] = class_id
+                    new_ann_instance['class_name'] = class_name
+                    if 'segmentation' in ann.keys():
+                        segmentation = ann['segmentation']
+                        if isinstance(segmentation, list) and len(segmentation) > 0: # polygon format
+                            new_ann_instance['segmentation'] = {'type':'polygon', 'task':'instance', 'data':segmentation}
+                        elif isinstance(segmentation, dict) and len(segmentation['counts']) > 0:
+                            new_ann_instance['segmentation'] = {'type':'mask', 'task':'instance', 'data':segmentation}
+                    if 'bbox' in ann.keys():
+                        new_ann_instance['bbox'] = ann['bbox']
+                    if 'keypoints' in ann.keys():
+                        new_ann_instance['keypoints'] = ann['keypoints']
+
+                    new_ann[component_name]['annotations'].append(new_ann_instance)
+
+                self.add_data(source_name, {
+                    component_name: img,
+                    'json': new_ann
+                }, split=split)
+
+            else:
+                warnings.warn(f"skipping {fn} as it does no exist!")
+
+    def from_yolo(self, source_name, yaml_path, split='all', override_main_component=None):
```

Comment:
As we grow, might be good to move those to separate files. Something like "parsers". To keep the readability of dataset and to allow easy support of additional parsers.

Reply:
Split these into a new file